### PR TITLE
add shape check for random_samples in fractional_max_pool{2d|3d}

### PR DESCRIPTION
**Pull request by mingfeima**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #89992

This PR add shape checks for `random_samples` in fractional_max_pool2d and fractional_max_pool3d.,
to provide more meaningful warnings instead of SegFault when the input is illegal.

For more details, please check https://github.com/pytorch/pytorch/issues/89648

### CODE REVIEW

File: aten/src/ATen/native/FractionalMaxPooling.h [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+template <int64_t ndim>
+static inline void fractional_max_pool_check_shape(
+    const Tensor& input,
+    const Tensor& randomSamples) {
+
+  TORCH_CHECK(
+      input.scalar_type() == randomSamples.scalar_type(),
+      "Expect _random_samples to have the dtype as input");
```
Comment (jgong5) [inline aten/src/ATen/native/FractionalMaxPooling.h:37]:
```suggestion
      "Expect _random_samples to have the same dtype as input");
```

**Review by jgong5 — APPROVED** [review]
LGTM. A minor revision about an error message. Please check. Also, shall we add UT to cover this?

File: aten/src/ATen/native/FractionalMaxPooling.h (hunk 1 above)
Comment (mingfeima) [reply aten/src/ATen/native/FractionalMaxPooling.h]: fixed!

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**